lens-cube/src/main/java/org/apache/lens/cube/parse/StorageCandidate.java

```diff
@@ -518,13 +579,51 @@ private boolean evaluateMeasuresCompleteness(TimeRange timeRange) throws LensExc
   }
 
   @Override
-  public Set<FactPartition> getParticipatingPartitions() {
-    return null;
+  public boolean isExpressionEvaluable(ExpressionResolver.ExpressionContext expr) {
+    return expr.isEvaluable(this);
   }
 
   @Override
-  public boolean isExpressionEvaluable(ExpressionResolver.ExpressionContext expr) {
-    return expr.isEvaluable(this);
+  public void updateAnswerableQueriedColumns(CubeQueryContext cubeql) throws LensException {
```

Comment:
Should we call this  updateAnswerableSelectColumns ? Also do we need this only in StoargeCandidate or required for other Candidates also ?